src/System.Linq.Expressions/tests/ILReader/ILProvider.cs

```diff
@@ -0,0 +1,39 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// Code adapted from https://blogs.msdn.microsoft.com/haibo_luo/2010/04/19/ilvisualizer-2010-solution
+
+using System.Reflection;
+
+namespace System.Linq.Expressions.Tests
+{
+    public interface IILProvider
+    {
+        byte[] GetByteArray();
+    }
+
+#if ILREADER_METHODBASE
```

Comment:
We lack some members on MethodBase API surface to make this work, but we don't need it right now. Was initially leaning towards keeping it in case it comes handy, but we can always get it from the original source, so removing it now.